### ISSUE TITLE
Setup Sentry secrets on CI

### ISSUE THREAD

**Issue by igor-sirotin** [status-im/status-react#21706]
Sentry integration is done here: https://github.com/status-im/status-go/pull/6054

# Description

We want to keep [Sentry DSN](https://docs.sentry.io/concepts/key-terms/dsn-explainer/) private, so we need to provide some environment variables on CI.

The following env vars need to be defined on _PRODUCTION_ CI only. 
Not for PR-level or develop builds.

- set `SENTRY_DSN_STATUS_GO` to the DSN from [here](https://sentry.infra.status.im/settings/sentry/projects/status-go/keys/)
- set `SENTRY_PRODUCTION` to `true`

**Comment by jakubgs** [comment]
Depends on:

- https://github.com/status-im/status-mobile/pull/21682

So for now it's on hold.

**Comment by ilmotta** [comment]
> Depends on:
> 
> * [[DRAFT] Configure Sentry for Mobile #21682](https://github.com/status-im/status-mobile/pull/21682)
> 
> So for now it's on hold.

@jakubgs @markoburcul the implementation in PR https://github.com/status-im/status-mobile/pull/21682 should work because I tested it locally pointing to a cloud Sentry instance. So we can proceed with this issue.

In `status-desktop` the make target building status-go is passing two new env vars `SENTRY_CONTEXT_NAME` and `SENTRY_CONTEXT_VERSION`. We need to do the same to build status-go for mobile. Could you assist me in how to implement this in `status-go/mobile/build.nix`? Actually, I would greatly appreciate if you could do the implementation on the nix layer 😅 Please, let me know if this is reasonable on your end. Thanks in advance 🙏🏼 

The value for `SENTRY_CONTEXT_NAME` should be `status-mobile` and `SENTRY_CONTEXT_VERSION` should be whatever value is in the `VERSION` file, just as was implemented in status-desktop AFAIU.

Once this issue #21706 is resolved I will proceed to rebase `develop` onto https://github.com/status-im/status-mobile/pull/21682

-----

@igor-sirotin in status-desktop it appears that `SENTRY_ENVIRONMENT` is not being used, only the env var `SENTRY_PRODUCTION`. In status-mobile I opted to follow the [table documented in status-go]( https://github.com/status-im/status-go/blob/8a7f24b095e078544368336034b148ee95c1e775/internal/sentry/README.md#environment) to define the values of `SENTRY_ENVIRONMENT` for each `.env.*` file. Is that the idea or have I misunderstood something?

**Comment by markoburcul** [comment]
> Could you assist me in how to implement this in `status-go/mobile/build.nix`?

I don't see that file, do you want to create it or? Since in status-desktop as I can see those two env variables are passed in the [Makefile](https://github.com/status-im/status-desktop/blob/95333e4b4a95837a2b03dfedd07abe8fafd16d22/Makefile#L442C1-L443C48).

**Comment by ilmotta** [comment]
> > Could you assist me in how to implement this in `status-go/mobile/build.nix`?
> 
> I don't see that file, do you want to create it or? Since in status-desktop as I can see those two env variables are passed in the [Makefile](https://github.com/status-im/status-desktop/blob/95333e4b4a95837a2b03dfedd07abe8fafd16d22/Makefile#L442C1-L443C48).

Thanks @markoburcul. The file is in this status-mobile repo at `nix/status-go/mobile/build.nix`, but when I copied I accidentally removed the parent path `nix/`.

**Comment by markoburcul** [comment]
> Thanks [@markoburcul](https://github.com/markoburcul). The file is in this status-mobile repo at `nix/status-go/mobile/build.nix`, but when I copied I accidentally removed the parent path `nix/`.

could you give me a link to this file on Github?

**Comment by ilmotta** [comment]
> > Thanks [@markoburcul](https://github.com/markoburcul). The file is in this status-mobile repo at `nix/status-go/mobile/build.nix`, but when I copied I accidentally removed the parent path `nix/`.
> 
> could you give me a link to this file on Github?

@markoburcul it's this file https://github.com/status-im/status-mobile/blob/d0deb94bf3f051cdb3a255c58faf03a1d7db0f31/nix/status-go/mobile/build.nix#L59

**Comment by igor-sirotin** [comment]
> in status-desktop it appears that SENTRY_ENVIRONMENT is not being used, only the env var SENTRY_PRODUCTION. 

@ilmotta indeed, yes. I didn't bother with other environments for now, because we don't yet report to sentry from e2e tests.
I'll do the same later when we will set it up for tests. 

> I opted to follow the [table documented in status-go](https://github.com/status-im/status-go/blob/8a7f24b095e078544368336034b148ee95c1e775/internal/sentry/README.md#environment) 

❤

**Comment by markoburcul** [comment]
@ilmotta I assume we finished the integration then and can close the task?

**Comment by ilmotta** [comment]
> [@ilmotta](https://github.com/ilmotta) I assume we finished the integration then and can close the task?

Yes, please @markoburcul

**Comment by igor-sirotin** [comment]
I guess it's done now 👌
